### PR TITLE
Add ClientOptions.Default

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

File: sdk/core/Azure.Core/src/ClientOptions.cs [hunk 1]
```diff
@@ -13,15 +13,66 @@ namespace Azure.Core
...
+                    IsLoggingContentEnabled = Default.Diagnostics.IsLoggingContentEnabled
+                };
+
+                _transport = Default.Transport;
+                PerCallPolicies = new List<HttpPipelinePolicy>(Default.PerCallPolicies);
+                PerRetryPolicies = new List<HttpPipelinePolicy>(Default.PerRetryPolicies);
+            }
+            else
+            {
+                // Intentionally laving this null. The only consumer of this branch is
```
Comment (AlexGhiondea) [inline sdk/core/Azure.Core/src/ClientOptions.cs:68]: typo - laving -> leaving

File: sdk/core/Azure.Core/src/ClientOptions.cs [hunk 2]
```diff
@@ -13,15 +13,66 @@ namespace Azure.Core
...
+            if (useDefaults)
+            {
+                Retry = new RetryOptions
+                {
+                    MaxRetries = Default.Retry.MaxRetries,
+                    Delay = Default.Retry.Delay,
+                    MaxDelay = Default.Retry.MaxDelay,
+                    Mode = Default.Retry.Mode,
+                    NetworkTimeout = Default.Retry.NetworkTimeout
+                };
```
Comment (tg-msft) [inline sdk/core/Azure.Core/src/ClientOptions.cs:48]: Nit - I wonder if we should make copy .ctors on those types so we can say `new RetryOptions(Default.Retry)`.  I always feel like it's harder to forget to clone a property with a copy .ctor that's right next to all the property declarations.

**Review by tg-msft — APPROVED** [review]
Looks great.

File: sdk/core/Azure.Core/src/ClientOptions.cs (hunk 2 above)
Comment (pakrym) [reply sdk/core/Azure.Core/src/ClientOptions.cs]: Great idea!

**Pull request merged**